java/ql/src/utils/FlowTestCaseSupportMethods.qll

```diff
@@ -49,4 +49,304 @@ string contentToken(Content c) {
 /**
  * Returns the `content` wrapped by `component`, if any.
  */
-Content getContent(SummaryComponent component) { component = SummaryComponent::content(result) }
+private Content getContent(SummaryComponent component) {
+  component = SummaryComponent::content(result)
+}
+
+module SupportMethod {
+  GenMethod genMethodForContent(SummaryComponentStack c) {
+    result = min(GenMethod g | g.appliesTo(any(VoidType v), getContent(c.head())))
+  }
+
+  GenMethod genMethodFor(Type t, SummaryComponentStack c) {
+    result = min(GenMethod g | g.appliesTo(t, getContent(c.head())))
+  }
+
+  GetMethod getMethodForContent(SummaryComponentStack c) {
+    result = min(GetMethod g | g.appliesTo(any(VoidType v), getContent(c.head())))
+  }
+
+  GetMethod getMethodFor(Type t, SummaryComponentStack c) {
+    result = min(GetMethod g | g.appliesTo(t, getContent(c.head())))
+  }
+}
+
+bindingset[this]
+abstract class SupportMethod extends string {
+  abstract predicate appliesTo(Type t, Content c);
+
+  string getARequiredImport() { none() }
+
+  string getDefinition() { none() }
+
+  bindingset[this, arg]
+  abstract string getCall(string arg);
+
+  /**
+   * Gets the CSV row describing this support method if it is needed to set up the output for this test.
+   *
+   * For example, `newWithMapValue` will propagate a value from `Argument[0]` to `MapValue of ReturnValue`, and `getMapValue`
+   * will do the opposite.
+   */
+  string getCsvModel() { none() }
+}
+
+class SourceMethod extends SupportMethod {
+  SourceMethod() { this = "source" }
+
+  override predicate appliesTo(Type t, Content c) { none() }
+
+  bindingset[arg]
+  override string getCall(string arg) {
+    result = "source()" and
+    // suppress unused variable warning
+    arg = arg
+  }
+
+  override string getDefinition() { result = "Object source() { return null; }" }
+}
+
+class SinkMethod extends SupportMethod {
+  SinkMethod() { this = "sink" }
+
+  override predicate appliesTo(Type t, Content c) { none() }
+
+  bindingset[arg]
+  override string getCall(string arg) { result = "sink(" + arg + ")" }
+
+  override string getDefinition() { result = "void sink(Object o) { }" }
+}
+
+bindingset[this]
+abstract class GetMethod extends SupportMethod { }
+
+private class DefaultGetMethod extends GetMethod {
+  Content c;
+
+  // prefix with zzz so the default getter is always last
+  DefaultGetMethod() { this = "zzzDefaultGet" + contentToken(c) }
+
+  string getName() { result = "get" + contentToken(c) }
+
+  override predicate appliesTo(Type t, Content c1) {
+    c = c1 and
+    // suppress unused variable warning
```

Comment:
Do some manual magic to avoid a huge multiple here, e.g. `t = any(TestCase tc).getOutputType()`?